packages/teko/src/Teko_HeuristicBlockPermutation.cpp

```diff
@@ -636,12 +651,13 @@ std::tuple<Teuchos::RCP<Teuchos::ParameterList>, std::string> default_merged_sol
   const auto gmresTol = 0.9 * targetResidualReduction;
   {
     int maxIterations = 30;
-#ifdef TFTK_ENABLE_GPU
-    // Sub-block preconditioner construction costs can become dominant on the GPU,
-    // especially for DD-ILU based approaches.
-    // Bumping up the maximum numer of GMRES iterations is therefore warranted.
-    maxIterations = 50;
-#endif
+    if (is_gpu_build()) {
+      // Sub-block preconditioner construction costs can become dominant on the GPU,
+      // especially for DD-ILU based approaches.
+      // Bumping up the maximum numer of GMRES iterations is therefore warranted.
+      maxIterations = 50;
```

Comment:
How important is it to have this here in the defaults, instead of the user input deck?